app/javascript/components/visualization/DotPlot.js

```diff
@@ -0,0 +1,141 @@
+      expressionValuesURL,
+      annotationCellValuesURL,
+      annotation,
+      '',
+      450
+    )
+    plotEvent.complete()
+  }, [expressionValuesURL, annotationCellValuesURL, annotation.name, annotation.scope])
+  return (
+    <div>
+      <div id={graphId} className="dotplot-graph"></div>
+      <DotPlotLegend/>
+    </div>
+  )
+}
+
+/** Render Morpheus dot plot */
+function renderDotPlot(target, dataPath, annotPath, annotation, fitType='', dotHeight=450) {
+
+  $(target).empty()
+
+  // Collapse by mean
+  const tools = [{
+    name: 'Collapse',
+    params: {
+      collapse_method: 'Mean',
+      shape: 'circle',
+      collapse: ['Columns'],
+      collapse_to_fields: [annotation.name],
+      pass_expression: '>',
+      pass_value: '0',
+      percentile: '100',
+      compute_percent: true
+    }
+  }]
+
+  const config = {
+    shape: 'circle',
+    dataset: dataPath,
+    el: $(target),
+    menu: null,
+    colorScheme: {
+      scalingMode: 'relative'
+    },
+    focus: null,
+    // We implement our own trivial tab manager as it seems to be the only way
+    // (after 2+ hours of digging) to prevent morpheus auto-scrolling
+    // to the heatmap once it's rendered
+    tabManager: {
+      add: (options) => {
+        $(target).empty()
+        $(target).append(options.$el)
+        return {id: $(target).attr('id'), $panel: $(target)}
+      },
+      setTabTitle: () => {},
+      setActiveTab: () => {},
+      getWidth: () => $(target).width(),
+      getHeight: () => $(target).height(),
+      getTabCount: () => 1
+    },
+    tools
+  }
+
+  // Set height if specified, otherwise use default setting of 500 px
+  if (dotHeight !== undefined) {
+    config.height = dotHeight
+  } else {
+    config.height = 500
+  }
+
+  // Fit rows, columns, or both to screen
+  if (fitType === 'cols') {
+    config.columnSize = 'fit'
+  } else if (fitType === 'rows') {
+    config.rowSize = 'fit'
+  } else if (fitType === 'both') {
+    config.columnSize = 'fit'
+    config.rowSize = 'fit'
+  } else {
+    config.columnSize = null
+    config.rowSize = null
+  }
+
+  // Load annotations if specified
+  if (annotPath !== '') {
+    config.columnAnnotations = [{
+      file: annotPath,
+      datasetField: 'id',
+      fileField: 'NAME',
+      include: [annotation.name]
+    }]
+    config.columnSortBy = [
+      { field: annotation.name, order: 0 }
+    ]
+    config.columns = [
+      { field: annotation.name, display: 'text' }
+    ]
+    config.rows = [
+      { field: 'id', display: 'text' }
+    ]
+
+    // Create mapping of selected annotations to colorBrewer colors
+    const annotColorModel = {}
+    annotColorModel[annotation.name] = {}
+    const sortedAnnots = annotation['values'].sort()
+
+    // Calling % 27 will always return to the beginning of colorBrewerSet
+    // once we use all 27 values
+    $(sortedAnnots).each((index, annot) => {
+      annotColorModel[annotation.name][annot] = window.colorBrewerSet[index % 27]
```

Comment:
Better: use `getColorBrewerColor(index)`.  See example in `study-overview.js`, or `scatter-plot.js` as of #829.